### PR TITLE
Add an e2e tracing test

### PR DESCRIPTION
**Pull request by skhalash**
**Description**

Changes proposed in this pull request:

- Add a pkg with Gomega matchers that help search in an OTLP JSON file
- Add an e2e tracing test

**Related issue(s)**
https://github.com/kyma-project/telemetry-manager/issues/57

### CODE REVIEW

File: internal/otelmatchers/matchers.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
+package otelmatchers
+
+import (
+	"fmt"
+	"github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+func ConsistOfSpansWithIDs(spanIDs []string) types.GomegaMatcher {
```
Comment (dennis-ge) [inline internal/otelmatchers/matchers.go:10]: I think the names could be more descriptive. The name`spanIDs` is used 2 times (in the `WithTransform` and here) with different meanings. Here it describes the expected Span IDs and in the `WithTransform` it describes the actual Span IDs.  So something like `expectedSpanIDs` and `actualSpanIDs` would help understanding the code :)

File: internal/otelmatchers/matchers.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+		var spanIDs []string
+		for _, span := range spans {
+			spanIDs = append(spanIDs, span.SpanID)
+		}
+		return spanIDs, nil
+	}, gomega.ConsistOf(spanIDs))
+}
+
+func EachHaveTraceID(traceID string) types.GomegaMatcher {
```
Comment (dennis-ge) [inline internal/otelmatchers/matchers.go:30]: similar to above with `traceID`

**Review by dennis-ge — CHANGES_REQUESTED** [review]

File: internal/otelmatchers/matchers.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+		}
+		return actualSpanIDs, nil
+	}, gomega.ConsistOf(expectedSpanIDs))
+}
+
+func EachHaveTraceID(expectedTraceID string) types.GomegaMatcher {
+	return gomega.WithTransform(func(actual interface{}) ([]string, error) {
+		actualBytes, ok := actual.([]byte)
+		if !ok {
+			return nil, fmt.Errorf("ConsistOfSpansWithIDs rqquires a []byte, but got %T", actual)
```
Comment (dennis-ge) [inline internal/otelmatchers/matchers.go:34]: is this the wrong error message? it names `ConsistOfSpansWithIDs` in the `EachHaveTraceID` test

File: internal/otelmatchers/matchers.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		}
+
+		actualSpans, err := unmarshalOTLPJSONTraceData(actualBytes)
+		if err != nil {
+			return nil, fmt.Errorf("ConsistOfSpansWithIDs requires a valid OTLP JSON document: %v", err)
```
Comment (dennis-ge) [inline internal/otelmatchers/matchers.go:39]: same as above

Comment (skhalash) [reply internal/otelmatchers/matchers.go]: Fixed

File: internal/otelmatchers/matchers.go (hunk 3 above)
Comment (skhalash) [reply internal/otelmatchers/matchers.go]: Fixed

File: internal/otelmatchers/matchers.go (hunk 1 above)
Comment (skhalash) [reply internal/otelmatchers/matchers.go]: Agree, done

File: internal/otelmatchers/matchers.go (hunk 2 above)
Comment (skhalash) [reply internal/otelmatchers/matchers.go]: Done

**Review by dennis-ge — APPROVED** [review]

File: internal/otelmatchers/matchers.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
+package otelmatchers
```
[the remark this answers is not in the document]
Comment (skhalash) [reply internal/otelmatchers/matchers.go:1]: It's actually not uncommon to extract test utilities into a separate pkg, I just wanted to have that code isolated

**Pull request merged**